### PR TITLE
Added anti-affinity policy for pods

### PR DESCRIPTION
**Pull request by garethahealy**
#### What is this PR About?
https://github.com/redhat-cop/rego-policies/issues/41

#### Check list
- [x] Have you created a test case in `_test/conftest-unittests.sh`
- [x] Have you created a test case in `_test/opa-profile.sh`
- [x] Have you created a test case in `_test/gatekeeper-integrationtests.sh`
- [x] Have you followed the TESTING.md doc? If not, please provide commands/steps to test this PR.
- [x] Have you re-generated `POLICIES.md`

cc: @redhat-cop/rego-policies

### CODE REVIEW

File: _test/all-namespaces/ocp/bestpractices/all.yml [hunk 1]
```diff
@@ -84,6 +84,16 @@ objects:
...
+    affinity:
+      podAntiAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                    - Foo
+            topologyKey: failure-domain.beta.kubernetes.io/zone
```
Comment (sabre1041) [inline _test/all-namespaces/ocp/bestpractices/all.yml:96]: Can `topology.kubernetes.io/zone` be used instead as the old key is deprecated

https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#failure-domainbetakubernetesiozone

File: _test/all-namespaces/ocp/bestpractices/all.yml [hunk 2]
```diff
@@ -159,6 +169,16 @@ objects:
...
```
Comment (sabre1041) [inline _test/all-namespaces/ocp/bestpractices/all.yml:181]: Same as above

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Small change requested

File: _test/all-namespaces/ocp/bestpractices/all.yml (hunk 1 above)
Comment (garethahealy) [reply _test/all-namespaces/ocp/bestpractices/all.yml]: done

File: _test/all-namespaces/ocp/bestpractices/all.yml (hunk 2 above)
Comment (garethahealy) [reply _test/all-namespaces/ocp/bestpractices/all.yml]: done

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**